src/jira/client/jira-client.ts

```diff
@@ -393,12 +395,12 @@ const dedupCommits = (commits: JiraCommit[] = []): JiraCommit[] =>
 /**
  * Deduplicates issueKeys field for branches and commits
  */
-const dedupIssueKeys = (repositoryObj, logger?) => {
-	updateRepositoryIssueKeys(repositoryObj, logger);
+const dedupIssueKeys = (repositoryObj) => {
+	updateRepositoryIssueKeys(repositoryObj, uniq);
 };
 
 /**
- * Truncates branches and commits to first 100 issue keys for branch or commit
+ * Truncates branches, commits and PRs to their first 100 issue keys
  */
 const truncateIssueKeys = (repositoryObj) => {
 	updateRepositoryIssueKeys(repositoryObj, truncate);
```

Comment:
How does this line currently work in main? It is passing in a `truncate` function, but inside `updateRepositoryIssueKeys` calls it `logger` (which you removed in this PR) and never use it.

----

So is it true that the truncate never work?

Reply:
The `updateRepositoryIssueKeys` should always receive a `mutatingFunc` in its second parameter. You're completely right, in main we were just receiving a logger which is not really what we wanted here. I caught that when writing my test and fixed it.